### ISSUE TITLE
Typescript definition for moment.min and moment.max are incomplete

### ISSUE THREAD

**Issue by veliancreate** [moment/moment#3577]
**Description of the Issue and Steps to Reproduce:**

Moment.js Typescript definition of moment.max, and moment.min are incorrect (if the examples in the docs are correct). According to the examples in the docs, the definition of each function should take an expanded array or just an array

  `export function min(...moments: Moment[]): Moment;`
  `export function min(moments: Moment[]):Moment;`
  `export function max(...moments: Moment[]): Moment;`
  `export function max(moments:Moment[]):Moment;`

The[ Definitely Typed definitions](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/637e7d6df755e785387d5269cb9287cdc51b8cb7/moment/moment.d.ts#L663) include these.

**Comment by mj1856** [comment]
I believe this is already covered in #3419.  We know the TS needs work, and will get there eventually.  Thanks.

**Comment by veliancreate** [comment]
Thanks @mj1856. 

Just thought I'd let you know that I encountered another issue with the Moment.UnitOfTime type. Bizzarely my IDE (Visual Studio Code) picked up that type when importing `* as moment from "moment"`, but the build failed using angulr-cli. 

I had to just create that specific type and import it within the scope of my own angular code 

`type momentUnitOfTime = (
    "year" | "years" | "y" |
    "month" | "months" | "M" |
    "week" | "weeks" | "w" |
    "day" | "days" | "d" |
    "hour" | "hours" | "h" |
    "minute" | "minutes" | "m" |
    "second" | "seconds" | "s" |
    "millisecond" | "milliseconds" | "ms"
);`

which I copied from moment.d.ts. Moment.diff() requires this type to execute.
